security/crypto/bls12/bls12.go

```diff
@@ -217,46 +216,40 @@ func (bls *bls12Base) coreVerify(pubKey *PublicKey, message []byte, signature *b
 	return engine.Result().IsOne()
 }
 
-func (bls *bls12Base) popProve() *bls12.PointG2 {
+func (bls *bls12Base) popProve() (*bls12.PointG2, error) {
 	pubKey := bls.privateKey().Public().(*PublicKey)
 	proof, err := bls.coreSign(pubKey.ToBytes(), domainPOP)
 	if err != nil {
-		bls.logger.Panicf("Failed to generate proof-of-possession: %v", err)
+		return nil, fmt.Errorf("Failed to generate proof-of-possession: %v", err)
```

Comment:
lower case: failed ... and use the `%w` verb instead of `%v` when adding `err` at the end like this, because then the root cause error can be unwrapped from the returned error. (see the std lib errors package for detailed docs.)

PS: This rule only applies to error messages, not logging.